src/subarrayn.jl

```diff
@@ -0,0 +1,25 @@
+const unsafe_getindex = Base.unsafe_getindex
+
+function Base.isnull{T,N,P<:NullableArray,IV,LD}(V::SubArray{T,N,P,IV,LD}, I::Int...)
+    @Base._inline_meta
+    @boundscheck checkbounds(V, I...)
+    @inbounds r = V.parent.isnull[Base.reindex(V, V.indexes, Base.to_indexes(I...))...]
+    r
+end
+
+function Base.values{T,N,P<:NullableArray,IV,LD}(V::SubArray{T,N,P,IV,LD}, I::Int...)
+    @Base._inline_meta
+    @boundscheck checkbounds(V, I...)
+    @inbounds r = V.parent.values[Base.reindex(V, V.indexes, Base.to_indexes(I...))...]
+    r
+end
+
+@generated function anynull{T, N, U<:NullableArray}(S::SubArray{T, N, U})
```

Comment:
It gives strange error messages under v0.4.5 (fine under v0.5) when I try to combine them:
```julia
ERROR: LoadError: LoadError: wrong number of arguments
 in include at /Applications/Julia-0.4.5.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include_from_node1 at /Applications/Julia-0.4.5.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include at /Applications/Julia-0.4.5.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include_from_node1 at /Applications/Julia-0.4.5.app/Contents/Resources/julia/lib/julia/sys.dylib
 in require at /Applications/Julia-0.4.5.app/Contents/Resources/julia/lib/julia/sys.dylib
while loading /Users/scott/.julia/v0.4/NullableArrays/src/subarray.jl, in expression starting on line 102
while loading /Users/scott/.julia/v0.4/NullableArrays/src/NullableArrays.jl, in expression starting on line 33
```
The error corresponds to the line after the `end` from the `if VERSION < v"0.5.0-dev+2718"` ... `else` ... `end`.
I have no idea what is causing this issue in v0.4.5, a bug in the compiler maybe?